workloads/virtual-machines/vmctl.md

```diff
@@ -0,0 +1,144 @@
+# VMCTL
+
+## Background
+
+One remarkable difference between KubeVirt and other solutions that run Virtual
+Machine workloads in a container is the toplevel API. KubeVirt treats
+VirtualMachineInstances as a first class citizen by designating
+CustomResourceDefinitions to map/track VirtualMachine settings and attributes.
+This has considerable advantages, but there's a trade-off: native Kubernetes
+behavior for constructs such as Deployments, ReplicaSets, DaemonSets,
+StatefulSets or even not-yet-existing resources are designed to work directly
+with Pods. Because VirtualMachine and VirtualMachineInstance resources are
+simply defined outside the scope of Kubernetes responsibility, it will always
+be up to the KubeVirt project to create analogues of those controllers. This is
+possible, and is in fact something that exists for some entities, e.g.
+VirtualMachineInstanceReplicaSet, but the KubeVirt project will always be one
+step behind. Any significant changes upstream would need to be implemented
+manually in KubeVirt.
+
+## Overview
+
+Vmctl is designed to address this delta by managing VirtualMachines from within
+a Pod. Vmctl will take an upstream VirtualMachine to act as a prototype and
+derive/spawn a new VirtualMachine based on it. This approach allows an
+arbitrary number of VirtualMachines that are nearly identical to be spawned
+using Kubernetes-native constructs. Each VirtualMachine is an exact copy of the
```

Comment:
I really like the paragraph until this point.

Maybe it's worth digging a little more into what this approach enables, something alogn the lines of:

> vmctl is running in a pod. The VM it dervied and spawned, will be runing alongside of it. Thus for every vmctl pod in the cluster, there should be a Vm running alongside of it. To be clear, vmctl _is not a VM_ instead it is controlling a VM close by.